### DISCUSSION TITLE
Markdown to HTML and vice versa

### DISCUSSION THREAD

**Discussion by peter-oms** [unifiedjs/unified#208] [Q&A] [answered]
Hi,

This is perhaps a question for @wooorm as I have been reading the ["HTML and remark" recipe page](https://unifiedjs.com/learn/recipe/remark-html/) and I saw your name against it.

Using the recipe code, convering HTML to Markdown and then converting the resulting Markdown back to HTML does not result in the original HTML.

[See this CodeSandbox for an example of this behaviour](https://codesandbox.io/s/fast-leaf-zbqrs7?file=/src/index.js) - starting with the HTML string `<p><strong>Hello </strong>world</p>`, I end up with `<p>**Hello **world</p>` after the two conversions.

I think the problem comes from converting the initial HTML to Markdown - I end up with `**Hello **world`, but this is probably not valid Markdown (space inside the bold tag), so when converting back to HTML it does not think it should add the `<strong>` tag.

I couldn't see any relevant options that could be applied to any of the rehype or remark packages to fix this behaviour. Have you had a similar issue before? Any ideas how to fix it?

**Comment by wooorm** [comment, marked as answer]
The problem is that markdown can’t really represent that HTML. As you noted. There is a nice way around it: improve the HTML you have. Don’t put that space there.

There is one automatic way around it, but it is extremely ugly. It’s to turn:

```html
<strong>Hello </strong>world
```

Into

```markdown
**Hello&#x20;**&#x77;orld
```

Which renders as: **Hello&#x20;**&#x77;orld

I’ve laid the solution out in https://github.com/orgs/syntax-tree/discussions/60#discussioncomment-2111096. It’s quite complex to wrap your head around though. Solving it would go in https://github.com/syntax-tree/mdast-util-to-markdown.
